server/src/main/scala/cromwell/CommandLineArguments.scala

```diff
@@ -60,26 +60,34 @@ case class CommandLineArguments(command: Option[Command] = None,
   }
 
   def validateSubmission(logger: Logger): ErrorOr[ValidSubmission] = {
-    val workflowPath = File(workflowSource.get.pathAsString)
+    def preProcessCwlWorkflowSource(workflowSourcePath: Path): ErrorOr[String] = {
+      val workflowPath = File(workflowSourcePath.pathAsString)
 
-    val workflowAndDependencies: ErrorOr[(String, Option[File], Option[String])] = if (isCwl) {
       logger.info("Pre Processing Workflow...")
       lazy val preProcessedCwl = cwlPreProcessor.preProcessCwlFileToString(workflowPath, None)
 
       imports match {
-        case Some(explicitImports) => readOptionContent("Workflow source", workflowSource).map((_, Option(File(explicitImports.pathAsString)), workflowRoot))
-        case None => Try(preProcessedCwl.map((_, None, None)).value.unsafeRunSync())
+        case Some(_) => readContent("Workflow source", workflowSourcePath)
+        case None => Try(preProcessedCwl.value.unsafeRunSync())
           .toChecked
           .flatMap(identity)
           .toValidated
       }
-    } else readOptionContent("Workflow source", workflowSource).map((_, imports.map(p => File(p.pathAsString)), workflowRoot))
+    }
 
-    val workflowSourceFinal: ErrorOr[String] = (workflowSource, workflowUrl) match {
-      case (Some(path), None) => readContent("Workflow source", path)
-      case (None, Some(url)) =>  PartialWorkflowSources.convertStringToUrl(url)
-      case (Some(_), Some(_)) => "Both Workflow source and Workflow url can't be supplied".invalidNel
-      case (None, None) => "Workflow source and Workflow url needs to be supplied".invalidNel
+    def getWorkflowSourceFromPath(workflowPath: Path): ErrorOr[(Option[String], Option[String])] = {
```

Comment:
TOL This 2-tuple of optional Strings would probably be better as a proper type, something like

```
case class SourceOrUrl(source: Option[String], url: Option[String])
```

but there are other options for representing this as well.